src/common-utils/main.sh

```diff
@@ -543,8 +544,14 @@ if [ "${INSTALL_ZSH}" = "true" ]; then
 
         # Add devcontainer .zshrc template
         if [ "$INSTALL_OH_MY_ZSH_CONFIG" = "true" ]; then
+            if [ -f "$user_rc_file" ]; then
+                cp -f ${user_rc_file} ${copy_user_rc_file}
+            fi
             echo -e "$(cat "${template_path}")\nDISABLE_AUTO_UPDATE=true\nDISABLE_UPDATE_PROMPT=true" > ${user_rc_file}
             sed -i -e 's/ZSH_THEME=.*/ZSH_THEME="devcontainers"/g' ${user_rc_file}
+            if [ -f "$user_rc_file" ]; then
+                diff ${user_rc_file} ${copy_user_rc_file} | grep '^>' | sed 's/^>\ //' >> ${user_rc_file}
```

Comment:
I don't think we should use `diff` as it has potential drawbacks 👇 

<img width="617" alt="image" src="https://github.com/user-attachments/assets/7093fc64-27e4-4c32-a3db-27f7b231d4a8">

Are there any concerns in appending the new contents to the end of the existing file?

Reply:
@samruddhikhandale 
There is no concern appending a new contents to the end of the existing file , the concern is, when user is appending a content to the existing file `~/.zshrc` the file is overwritten by the zshrc_template file and all the content added by user is getting disappeared from the file. 

before [this](https://github.com/devcontainers/features/pull/1069/files/6f61ee04b67b8f8bdb382455c8b4e607f12e5633#diff-3e71c0a0669a0410f7dd0d8f2b83f3b6bf6b525d3eabd354f19f32822da669fcR550) the user_rc_file has user provided content already appended, but after this it is overwritten by the template file.

can you suggest alternate option if diff is not recommended

Reply:
I have introduced a new logic without affecting the existing logic.
now, if the `.zshrc` file exist then we will just update the necessary attributes from the file and if the named file is not exist then only we will create a file with the default zshrc-template file
hope this logic will work as expected. Kindly review it once

Reply:
Yep, makes sense! Looking